### ISSUE TITLE
[Backend] As ACF staff, I must use ACF AMS for authentication.

### ISSUE THREAD

**Issue by abottoms-coder** [raft-tech/TANF-app#1170]
**Description:**
_Provide a brief background and justification for this issue_
ACF users (OFA sys admin, OFA admin, OFA Regional Staff, OFA Data Analyst) should not be allowed to utilize anything except ACF AMS.

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [x] ACF staff user is not actually authenticated by Login.gov (not able to browse/utilize TDP)
- [x] If an ACF staff user uses  login.gov for authentication,  an error is returned indicating incorrect authentication method
- [x] First-time ACF user story is handled for a user not yet assigned a role
- [x] Testing Checklist has been run and all tests pass
- [x] Ensure logout is working as expected
- [x] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 
- [x] From openid payload, trigger off @acf.hhs.gov e-mail domain for determining if failure is warranted
- [x] Make sure logout redirects as expected
- [x] Run Testing Checklist and confirm all tests pass

**Notes:**
_Add additional useful information, such as related issues and functionality that isn't covered by this specific issue, and other considerations that will be helpful for anyone reading this_ 

**Supporting Documentation:**
_Please include any relevant log snippets/files/screen shots_

**Comment by amilash** [comment]
@abottoms-coder can you add any relevant criteria/dev work to build for enforcing piv/cac into this ticket?

**Comment by amilash** [comment]
Blocked by design user flows

**Comment by valcollignon** [comment]
We need answers to open questions before we can refine. Per backlog refinement 11.2.21.

**Comment by lfrohlich** [comment]
@ADPennington  : how do you enforce PIV/CAC if they don't have a role, this is their first time logging in? Need to better understand this flow

**Comment by valcollignon** [comment]
Discuss this in dev sync to revisit open questions. Priority is #1136.

**Comment by ADPennington** [comment]
> Discuss this in dev sync to revisit open questions. Priority is #1136.

### Notes from dev sync:
- The expectation is that first time ACF user will request access to TDP after clicking ACF AMS button and authenticating via PIV/CAC. **_At this stage they will not have a role_**, so in order to confirm that the correct authentication button was used, we would ideally rely on metadata returned about the user that lets us know that this is an ACF user (e.g. has an acf.hhs.gov email address). 
- We will  confirm via #1136 the types of metadata to be returned about these users from ACF AMS. 
- if, based on the metadata, it is determined 1)this is an ACF user and 2) this user did not use ACF AMS to authenticate, an error would be returned to the user informing that the incorrect authentication method was used.

**Comment by valcollignon** [comment]
Keep in unrefined until issue is unblocked - per backlog refinement 11.16.21

**Comment by valcollignon** [comment]
Front end work is being done in #1155.

**Comment by valcollignon** [comment]
As soon as #1136 is complete, this issue may be brought into the current sprint immediately.
CC: @abottoms-coder @ADPennington @lfrohlich

**Comment by valcollignon** [comment]
Per @abottoms-coder's conversation with Leon, this issue is blocked until February at the earliest. The openid platform we use to connect to AMS can't be setup to pass through neither the hhs_id nor any PIV-relevant information.

**Comment by ADPennington** [comment]
notes from 12/21 pair w/ @jorgegonzalez and @abottoms-coder:
- ACF AMS team informed today that they have an **_unknown_** timeline for setting up to return the `hhs_id` in the payload for our use. this ID is a nice to have and is not considered a blocker for this epic (returning the user `email` is sufficient)
- a `null` value will be returned for `hhs_id` field for now. 
- (1) there currently is no PIV/CAC metadata returned with the user payload + (2) Andrew and I are not able to authenticate using our ACF AMS network  u/p credentials---both of which we expected for us to complete this work. This suggests that using ACF AMS effectively means that a user has authenticated with PIV/CAC (because no other credentials are accepted), but we need to confirm the expected behavior with ACF AMS team. 
- We also need to confirm ACF OCIO's intention for TDP's ACF users leveraging ACF AMS authentication service.  Is enough to know that the user came through ACF AMS, or must we also know that they came through ACF AMS _and_ used PIV/CAC? if it is the latter,  we need to know what user metadata returned in the payload will tell us that and how it will be made available to us. 
- Depending on the outcome of these follow-ups, #1170 issue and associated ACs may need to be revised. 

cc: @lfrohlich

**Comment by ADPennington** [comment]
From ACF AMS team 12/23:
- ACF OCIO supports authenticating with PIV/CAC or network credentials when using ACF AMS. 
-  if network credentials are unsuccessful, this is something that would need to be elevated to the attention of the HHS AMS team if it is something we need to support (it isn't). What is most important is that ACF users are using ACF AMS to authenticate and not login.gov. 
- ACF AMS team will be speaking with HHS AMS team sometime in the new year and will ask about whether or not PIV metadata will ever be made available to return in the payload. 

Other notes:
- it is worth finding an HHS, non-ACF users to help us test to see what is returned in the payload on them. we expect them to have a different email address (i.e. not ending in `@acf.hhs.gov`) but in the worse case scenario, maybe they arent blocked from authenticating and make it into TDP to request access. even if this does happen, their permissions would still need to be assigned in order to have access to the TDP, so the risk is low for this to cause problems for us (aside from the user management headache). 
- this means that the title of this ticket should not include piv/cac as the only authentication credential. we should add this clarification to the agenda for our next IPT in 2022. 
- `ACF staff user cannot authenticate via username and password in ACF AMS` should be removed as an AC in this ticket.

**Comment by abottoms-coder** [comment]
Reopened to address #1600 .
